src/App.tsx

```diff
@@ -0,0 +1,68 @@
+import { useNavigate } from "react-router-dom";
+import "./index.css";
+import "./App.css";
+
+import { useState, useRef } from "react";
+import Modal from "@/components/form/modal";
+import ControlledForm from "@/components/form/ControledForm";
+import UncontrolledForm from "@/components/form/UnControledForm";
+import { useAppSelector } from "@/app/appHook";
+
+import { CardList } from "@/features/user/CardList";
+
+const App: React.FC = () => {
+  const navigate = useNavigate();
+  const [isOpen, setIsOpen] = useState(false);
+  const [formType, setFormType] = useState<"control" | "uncontrol">("control");
+  const result = useAppSelector((state) => state.user);
+  const results = result.data;
+  const lastFocusedButton = useRef<HTMLButtonElement | null>(null);
+
+  const controlFormHandle = (e: React.MouseEvent<HTMLButtonElement>) => {
+    lastFocusedButton.current = e.currentTarget;
+    navigate("/control", { replace: true });
+    setFormType("control");
+    setIsOpen(true);
+  };
+
+  const unControlFormHandle = (e: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
Instead of `controlFormHandle` and `unControlFormHandle`, consider renaming these functions. While the current names hint at what they do, they can be a bit redundant and could be clearer about their purpose.

```suggestion
  const openControlledForm = (e: MouseEvent<HTMLButtonElement>) => {
    lastFocusedButton.current = e.currentTarget;
    navigate("/control", { replace: true });
    setFormType("control");
    setIsOpen(true);
  };

  const openUncontrolledForm = (e: MouseEvent<HTMLButtonElement>) => {
```